### ISSUE TITLE
Refreshing of the page when navigating thru the history

### ISSUE THREAD

**Issue by Daidalos117** [recruit-tech/recoil-sync-next#163]
Hi there,
firstly thanks for the nice plugin, it's exactly what I am looking for :)
When there new state in the url and for example I want to go back, the whole pages reloads, which is not very good ux and it's unnecessary really.

When I was manipulating query by myself, I didn't have the issue, I was changing the url as this:
`router.push({
      query: { ...router.query, param: 'value' },
    });
`
Would it be possible to rewrite like this in the library so the unnecessary reload is not there?
Thanks!

**Comment by koichik** [comment]
Thanks for your interest!

I tried your suggestion but it had no affect.
Actually, Next.js internally transforms the URL object into string.

Please refer to:
https://github.com/vercel/next.js/blob/2f744a1ae32a65f1362af0957e3c57f33d102914/packages/next/shared/lib/router/router.ts#L1178
and
https://github.com/vercel/next.js/blob/2f744a1ae32a65f1362af0957e3c57f33d102914/packages/next/shared/lib/router/router.ts#L304-L306

Thus I think your code would be equivalent to passing a string URL.
(Perhaps I am misunderstanding your meaning, please provide a working example repository if possible.)


If we could use a static Router (`import Router from 'next/router'`) instead of `useRouter`, we can avoid re-rendering, I think.
However, the current implementation needs `isReady`, so `useRouter` must be used

**Comment by Daidalos117** [comment]
Well I started to experiment a bit, I have created example repo here: https://github.com/Daidalos117/url-state
What I have found is that if I use the recoil sync, there is sadly no history. Sync is probably using just replaceUrl, which is something I don't want. My primary target is to have history with all the states, so the user can go back.
I think I will just need to write my own syncer.
I recorded small videos with differences between working
[https://drive.google.com/file/d/14gk7wTcy7OJWqDEUAdjaJtp4VFzxu4Jp/view?usp=sharing](Manul handling)
[https://drive.google.com/file/d/1g_LObOvoF39bommMCBeM4F6rf8SMo90Y/view?usp=sharing](Recoil sync)
